### PR TITLE
AdvisorCommand: Print a summary of how many packages are vulnerable

### PR DESCRIPTION
**Pull request by sschuberth**
Signed-off-by: Sebastian Schuberth <sschuberth@gmail.com>

### CODE REVIEW

File: cli/src/main/kotlin/commands/AdvisorCommand.kt [hunk 1]
```diff
@@ -136,6 +136,10 @@ class AdvisorCommand : OrtCommand(
             throw ProgramResult(1)
         }
 
+        val vulnerablePackageCount = advisorResults.getVulnerabilities().size
+        val totalPackageCount = ortResultOutput.getPackages().size
```
Comment (fviernau) [inline cli/src/main/kotlin/commands/AdvisorCommand.kt:140]: The advisor may not have queried excluded packages, so the value for `totalPcakageCount` may be mis-leading the user into thinking that all these packages have been queried for vulnerabilities. So, should `skipExcluded` be passed to `getPackages()` ?

Apart from that, I wonder if total package count does matter at all or whether it should be dropped.
IMO adding the amount of distinct vulnerabilites or non-distinct vulnerabilities would be more interresting than total package count.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/commands/AdvisorCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/AdvisorCommand.kt:140]: > So, should `skipExcluded` be passed to `getPackages()` ?

I've now simply aligned the wording with the one in the `AnalyzerCommand`: https://github.com/oss-review-toolkit/ort/blob/2e216452db7db7c984306ff1d4d4f875ca1f2067/cli/src/main/kotlin/commands/AnalyzerCommand.kt#L284

> Apart from that, I wonder if total package count does matter at all or whether it should be dropped.

I believe it's informative to know what share of your packages are vulnerable.

> IMO adding the amount of distinct vulnerabilites or non-distinct vulnerabilities would be more interresting

I'm now also mentioning the total number of vulnerabilities, so one can quickly see if there's more than one vulnerability per package.

File: cli/src/main/kotlin/commands/AdvisorCommand.kt [hunk 2]
```diff
@@ -136,6 +136,16 @@ class AdvisorCommand : OrtCommand(
             throw ProgramResult(1)
         }
 
+        with(advisorResults.getVulnerabilities()) {
+            val totalPackageCount = ortResultOutput.getPackages().size
```
Comment (fviernau) [inline cli/src/main/kotlin/commands/AdvisorCommand.kt:140]: this is counting excluded packages, so it contradicts the log message, see: OrtResult.getPackages(omitExcluded: Boolean = false)

**Review by fviernau — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/commands/AdvisorCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/AdvisorCommand.kt]: Phew, this was also broken for the `AnalyzerCommand`. Thanks, both fixed.

**Review by fviernau — APPROVED** [review]

**Pull request merged**